src/main/java/reposense/authorship/AuthorshipReporter.java

```diff
@@ -2,24 +2,42 @@
 
 import java.util.List;
 import java.util.Objects;
+import java.util.logging.Logger;
 import java.util.stream.Collectors;
 
 import reposense.authorship.model.AuthorshipSummary;
 import reposense.authorship.model.FileInfo;
 import reposense.authorship.model.FileResult;
 import reposense.model.RepoConfiguration;
+import reposense.system.LogsManager;
+
 
 /**
  * Generates the authorship summary data for each repository.
  */
 public class AuthorshipReporter {
+    private static final Logger logger = LogsManager.getLogger(AuthorshipReporter.class);
+
+    private static final int HIGH_NUMBER_LINES_THRESHOLD = 500000;
+    private static final String HIGH_NUMBER_LINES_MESSAGE = "There are a large number (%s) of text files to be "
+            + "analyzed, comprising a total of %s lines. As such, RepoSense may take a long time to generate the "
+            + "report. Performance may be improved by setting values for the \"File formats\" and "
+            + " \"Ignore Glob List\" columns in the repo-config.csv so as to reduce the number of files to be "
+            + " analyzed.";
 
     /**
      * Generates and returns the authorship summary for each repo in {@code config}.
      */
     public static AuthorshipSummary generateAuthorshipSummary(RepoConfiguration config) {
         List<FileInfo> textFileInfos = FileInfoExtractor.extractTextFileInfos(config);
 
+        int numFiles = textFileInfos.size();
+        int totalNumLines = textFileInfos.stream().mapToInt(fileInfo -> fileInfo.getNumOfLines()).sum();
```

Comment:
Great! This looks much neater now. Can you split this into multiple lines, like this?
```suggestion
        int totalNumLines = textFileInfos.stream()
                .mapToInt(fileInfo -> fileInfo.getNumOfLines())
                .sum();
```